pkg/collector/corechecks/cluster/kubernetes_state.go

```diff
@@ -270,26 +267,26 @@ func (k *KSMCheck) processMetrics(sender aggregator.Sender, metrics map[string][
 			if !isKnownMetric(metricFamily.Name) {
 				// ignore the metric if it doesn't have a transformer
 				// or if it isn't mapped to a datadog metric name
-				log.Tracef("KSM metric '%s' is unknown for the check, ignoring it", metricFamily.Name)
+				// log.Tracef("KSM metric '%s' is unknown for the check, ignoring it", metricFamily.Name)
 				continue
 			}
 			if transform, found := metricTransformers[metricFamily.Name]; found {
 				for _, m := range metricFamily.ListMetrics {
-					hostname, tags := k.hostnameAndTags(m.Labels, metricsToGet)
+					hostname, tags := k.hostnameAndTags(m.Labels, labelJoiner)
 					transform(sender, metricFamily.Name, m, hostname, tags)
 				}
 				continue
 			}
 			for _, m := range metricFamily.ListMetrics {
-				hostname, tags := k.hostnameAndTags(m.Labels, metricsToGet)
+				hostname, tags := k.hostnameAndTags(m.Labels, labelJoiner)
 				sender.Gauge(formatMetricName(metricFamily.Name), m.Val, hostname, tags)
 			}
 		}
 	}
 }
 
 // hostnameAndTags returns the tags and the hostname for a metric based on the metric labels and the check configuration
-func (k *KSMCheck) hostnameAndTags(labels map[string]string, metricsToGet []ksmstore.DDMetricsFam) (string, []string) {
+func (k *KSMCheck) hostnameAndTags(labels map[string]string, labelJoiner *labelJoiner) (string, []string) {
 	hostname := ""
 	tags := []string{}
```

Comment:
You could allocate capacity upfront to `len(labels) + len(labelsToAdd)` here, no?

Reply:
Yes I can! 👍🏻